src/build_index.rs

```diff
@@ -187,7 +187,7 @@ fn group_by_slices<T, K: PartialEq, F: Fn(&T) -> K>(
             slice_start = i;
         }
     }
-    if slice_start > 0 {
+    if slice_start > 0 || data.len() <= min_size {
```

Comment:
This is fixing a bug that if there were fewer total elements than slice_size, the first (partial) slice would not be included

Reply:
Would be good to add a unit test that fails without this fix.